### PR TITLE
If device is not supported, gracefully ignore it #184

### PR DESCRIPTION
**Pull request by elad-bar**
### **PR Type**
bug_fix

___

### **Description**
- Added a check to ensure `counter_devices` is not `None` before accessing its elements, preventing potential errors.
- Refactored the code to handle unsupported devices more gracefully by logging a warning and falling back to the Masa API.
- Improved the robustness of the `_estimate_bill` method by ensuring it does not fail when encountering unsupported devices.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>coordinator.py</strong><dd><code>Gracefully handle unsupported devices in billing estimation</code></dd></summary>
<hr>

custom_components/iec/coordinator.py

<li>Added a check for <code>counter_devices</code> being not <code>None</code> before accessing its <br>elements.<br> <li> Refactored code to handle unsupported devices gracefully.<br> <li> Improved error handling by logging a warning and falling back to an <br>alternative API.<br>

</details>

  </td>
  <td><a href="https://github.com/GuyKh/iec-custom-component/pull/185/files#diff-6ce6bbf13b6589ae1f4611633e1444dd2f21e3bf539c5cc80f3997b7ef9b40b5">+12/-5</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

**Comment by elad-bar** [comment]
test it locally, works fine

not sure if it works like that in the iec client, but each of the response from iec api has a block of status (reponseDescriptor.isSuccess), maybe have a warning log when it returns an issue will be easier to debug, 
if that's something you would like to have, i can try to do the and suggest another PR

```json
{
    "data": {
        "counterDevices": null,
        "mrType": "00"
    },
    "reponseDescriptor": {
        "isSuccess": false,
        "code": "08",
        "description": "No authorization for this product: PV Product"
    }
}
```

**Comment by elad-bar** [comment]
will fix the lint error

File: custom_components/iec/coordinator.py [hunk 1]
```diff
@@ -629,11 +629,18 @@ async def _estimate_bill(self, contract_id, device_number, is_private_producer,
...
-                last_meter_read = int(devices_by_id.counter_devices[0].last_mr)
-                last_meter_read_date = devices_by_id.counter_devices[0].last_mr_date
-                phase_count = devices_by_id.counter_devices[0].connection_size.phase
-                connection_size = (devices_by_id.counter_devices[0].
-                                    connection_size.representative_connection_size)
+
+                counter_devices = devices_by_id.counter_devices
+
+                if counter_devices is not None:
+                    first_counter_device = counter_devices[0]
```
Comment (GuyKh) [inline custom_components/iec/coordinator.py:636]: `counter_devices` could be also an empty list, in that case.
so it's better to write this as `if not counter_devices:`.

But the main question is if the data in masa (the fallback) is any better.

File: custom_components/iec/coordinator.py [hunk 2]
```diff
@@ -629,11 +629,18 @@ async def _estimate_bill(self, contract_id, device_number, is_private_producer,
...
+                    connection_size = first_counter_device.connection_size
```
Comment (GuyKh) [inline custom_components/iec/coordinator.py:637]: To be honest, I already had this in local branch.

```suggestion
                if devices_by_id.counter_devices and len(devices_by_id.counter_devices) >= 1:
                    last_meter_read = int(devices_by_id.counter_devices[0].last_mr)
                    last_meter_read_date = devices_by_id.counter_devices[0].last_mr_date
                    phase_count = devices_by_id.counter_devices[0].connection_size.phase
                    connection_size = (devices_by_id.counter_devices[0].
                                        connection_size.representative_connection_size)
                else:
                    _LOGGER.warning("Failed to get Last Device Meter Reading, trying another way...")
```

Comment (elad-bar) [reply custom_components/iec/coordinator.py:637]: Thanks!

**Comment by elad-bar** [comment]
Will wait for your release

**Pull request closed without merging**